### PR TITLE
[FCE-274] Remove enums from SDK

### PR DESCRIPTION
**Pull request by mironiasty**
## Description

Remove `enum`s from TypeScript code. And use `union`s instead. 

Fix bug, where `EncodingReason` had incorrect mapping to native part.

## Motivation and Context

Enums bad. Unions good.

### CODE REVIEW

File: examples/video-chat/hooks/useJoinRoom.ts
Comment (karkakol) [inline examples/video-chat/hooks/useJoinRoom.ts:1]: fix warning

File: examples/video-chat/screens/RoomScreen.tsx
Comment (karkakol) [inline examples/video-chat/screens/RoomScreen.tsx:1]: fix warning

**Review by karkakol — COMMENTED** [review]
It would be nice to unify types string casing. Once we use `export type VideoLayout = 'FILL' | 'FIT'`, in other place `export type VadStatus = 'silence' | 'speech';`. But I am not sure there exists eslint rule for it

**Comment by mironiasty** [comment]
> It would be nice to unify types string casing. Once we use export type VideoLayout = 'FILL' | 'FIT', in other place export type VadStatus = 'silence' | 'speech';. But I am not sure there exists eslint rule for it

Maybe, but I would prefer to do it in separate PR. This change would require native code changes + and much more testing

File: examples/video-chat/hooks/useJoinRoom.ts
Comment (mironiasty) [reply examples/video-chat/hooks/useJoinRoom.ts:1]: I've removed warnings + I've updated eslint config to fail in case of warnings in future.

**Review by karkakol — APPROVED** [review]

**Pull request merged**